### PR TITLE
Modified First Pull Request

### PR DESCRIPTION
**Pull request by Illegal32**

### CODE REVIEW

File: src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        studentEntity = studentRepository.save(studentEntity);
+
+        return studentService.generateStudentEmailAccount(studentEntity.getId(),
+                studentEntity.getName(),
+                studentEntity.getSurname());
+    }
+
+    @Override
+    public Iterable<StudentEntity> findAll() {
```
Comment (Illegal32) [inline src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java]: replace with List

File: src/main/groovy/az/pashabank/controller/StudentController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+public class StudentController {
+    private final StudentServiceImpl studentService;
+
+    public StudentController(StudentServiceImpl studentService) {
+        this.studentService = studentService;
+    }
+
+    @PostMapping
+    public ResponseEntity<StudentRequestDto> save(@RequestBody StudentRequestDto studentRequestDto) {
+        var result = studentService.save(studentRequestDto);
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/controller/StudentController.java]: why do you need ResponseEntity?)

File: src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        this.studentRepository = studentRepository;
+        studentMapper = StudentMapper.INSTANCE;
+    }
+
+    @Override
+    public String generateStudentEmailAccount(Long id, String name, String surname) {
+        String firstLetter = String.valueOf(name.charAt(0));
+        String email = firstLetter.concat(surname).concat(String.valueOf(id));
+
+        return email.concat(DOMAIN);
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java]: let's do this with just one line:

return String.valueOf(name.charAt(0)).concat(surname).concat(String.valueOf(id)).concat(DOMAIN);

File: src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    final String DOMAIN = "@pashabank.az";
+
+    private final StudentMapper studentMapper;
+
+    private final StudentRepository studentRepository;
+
+    public StudentServiceImpl(StudentRepository studentRepository) {
+        this.studentRepository = studentRepository;
+        studentMapper = StudentMapper.INSTANCE;
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/service/impl/StudentServiceImpl.java]: this.

File: src/main/groovy/az/pashabank/model/dto/StudentRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ * Time : 4:35 PM
+ * Project Name : student-service
+ */
+
+package az.pashabank.model.dto;
+
+import lombok.Data;
+
+@Data
+public class StudentRequestDto {
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/model/dto/StudentRequestDto.java:19]: why do you need StudentRequestDto?

File: src/main/groovy/az/pashabank/model/dto/StudentDto.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * Time : 2:29 PM
+ * Project Name : student-service
+ */
+
+package az.pashabank.model.dto;
+
+import lombok.Data;
+
+@Data
+public class StudentDto {
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/model/dto/StudentDto.java:19]: what's the difference between StudentDto and StudentEntity?

File: src/main/groovy/az/pashabank/model/entity/StudentEntity.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ * Project Name : student-service
+ */
+
+package az.pashabank.model.entity;
+
+import lombok.*;
+
+import javax.persistence.*;
+
+@Table(name = "student_info")
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/model/entity/StudentEntity.java]: it's better to set table name just 'student'. Same as entity

File: src/main/groovy/az/pashabank/service/StudentService.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import az.pashabank.model.dto.StudentDto;
+import az.pashabank.model.dto.StudentRequestDto;
+
+import java.util.List;
+
+public interface StudentService {
+
+    String generateStudentEmailAccount(Long id, String name, String surname);
+
+    StudentRequestDto save(StudentRequestDto studentRequestDto);
```
Comment (validakhundov) [inline src/main/groovy/az/pashabank/service/StudentService.java]: why save method returns StudentRequestDto?

**Review by validakhundov — CHANGES_REQUESTED** [review]
Please be careful about naming

**Pull request merged**